src/lua/utils.c

```diff
@@ -735,6 +735,22 @@ luaT_newthread(struct lua_State *L)
 	return L1;
 }
 
+/** A getenv wrapper checking for size of the corresponding env variable. */
+static int
+luaT_getenv_safe(struct lua_State *L)
+{
+	if (lua_gettop(L) != 2 || !lua_isstring(L, 1) || !lua_isnumber(L, 2))
+		return luaL_error(L, "usage: getenv_safe(name, length)");
+	const char *name = lua_tostring(L, 1);
+	int len = lua_tonumber(L, 2);
+	const char *env = getenv_safe(name, len, NULL);
+	if (env != NULL) {
+		lua_pushstring(L, env);
+		return 1;
+	}
+	return 0;
+}
```

Comment:
One more raw thought: shouldn't we do something with the `os.environ` table (in Lua)? Don't know what, but it has no any safe analogue. Don't know, whether it should do.

Reply:
Me neither. What would it look like? A version which returns truncated strings, probably?

I'd rather not bother with that in scope of this ticket.

Reply:
Okay. Let's ignore till someone will complain. We anyway have no vision of a proper solution.